packages/react-dom/src/events/BeforeInputEventPlugin.js

```diff
@@ -29,6 +28,7 @@ import {
 } from './FallbackCompositionState';
 import SyntheticCompositionEvent from './SyntheticCompositionEvent';
 import SyntheticInputEvent from './SyntheticInputEvent';
+import {accumulateTwoPhaseListeners} from './DOMModernPluginEventSystem';
```

Comment:
Why is shared code unconditionally calling an implementation from the modern event system? If it is standalone, should it have been shared? This seems error-prone.

Reply:
Also can we delete `accumulateTwoPhaseDispatchesSingle` and `traverseTwoPhase` now?

I would find it clearer if the new implementation moved to where they lived before. Given that it’s used in legacy system too now.

Reply:
We can’t delete them just yet. They’re exported from ReactDOM in the hidden object.